File: algorithms/go/reverse-integer.go
import "math"

func reverseInteger(num int) int {
	sign := 1
	ans := 0
	if num < 0 {
		sign = -1
		num *= -1
	}

	for num > 0 {

		tmp := num % 10

		ans = ans*10 + tmp

		num /= 10
	}

	ans = ans * sign

	if ans > math.MaxInt32 || ans < math.MinInt32 {
		return 0
	}
	return ans
}

File: algorithms/javascript/reverse-integer.js
function reverseInteger(num) {
    let ans = 0
    while (num) {
        ans = ans * 10 + num % 10
        num = ~~(num / 10)
    }

    if (ans >= (Math.pow(2,31) - 1) || ans <= -Math.pow(2,31)) {
        return 0
    }
    return ans
}